homeassistant/components/sensor/broadlink.py

```diff
@@ -110,23 +110,29 @@ def __init__(self, interval, ip_addr, mac_addr, timeout):
         self.data = None
         self._device = broadlink.a1((ip_addr, 80), mac_addr)
         self._device.timeout = timeout
+        self._schema = vol.Schema({
+            vol.Optional('temperature'): vol.Range(min=-50, max=150),
+            vol.Optional('humidity'): vol.Range(min=0, max=100),
+            vol.Optional('light'): vol.Any(0, 1, 2, 3),
+            vol.Optional('air_quality'): vol.Any(0, 1, 2, 3),
+            vol.Optional('noise'): vol.Any(0, 1, 2),
+            })
         self.update = Throttle(interval)(self._update)
         if not self._auth():
             _LOGGER.warning("Failed to connect to device.")
 
     def _update(self, retry=3):
         try:
             data = self._device.check_sensors_raw()
-            if (data is not None and data.get('humidity', 0) <= 100 and
-                    data.get('light', 0) in [0, 1, 2, 3] and
-                    data.get('air_quality', 0) in [0, 1, 2, 3] and
-                    data.get('noise', 0) in [0, 1, 2]):
-                self.data = data
+            if data is not None:
+                self.data = self._schema(data)
```

Comment:
Exactly. And it will retry a couple of times, just with the socket timeout, until eventually giving up and leaving the value as is, until its next update. Which is the intended behavior - or, at least, what I had in mind.